### ISSUE TITLE
taking long time while re-sync

### ISSUE THREAD

**Issue by ramnkl** [kamsar/Unicorn#58]
It is taking long time while sync with another sitecore instance. 

While serialize it took about few mins for 1 GB serialized item where as while re-sync it is taking longer time still in the process of 66% creating template items taking longer time. Am I missing any configuration. While sync is executing I can't even access the sitecore instance (sitecore login). 
After reset the IIS would able to login to sitecore instance.
sitecore  : 7.2
unicorn 3.0RC1 

Here is the detail screen shot
![unicornresync](https://cloud.githubusercontent.com/assets/5870863/10122844/ad3d57bc-64ec-11e5-9f3f-e2ccfab6ea71.PNG)

**Comment by suneillg** [comment]
I am experiencing similar issues when doing "Synch Default Configuration Now" in another instance of SItecore.
It syncs few items and get stuck. When i try to reload the I get "The remote host closed the connection. The error code is 0x800704CD" error. I need to refresh the app pool and start the process again. It takes me about 3-4 tries. 
Once all the items are synced, the subsequent syns runs fine. 
Please let us know if there are any configuration that need to be added.

Sitecore: 8 rev 150427
unicorn 3.0RC1

Thank you

**Comment by kamsar** [comment]
I've replicated this and it appears to be some kind of concurrency issue involving the Sitecore data access layer. I'll see if I can isolate it, but in the meantime as a workaround you can reduce the max concurrency in Unicorn.config to 1 or 2 threads, and that should avoid the issue.

**Comment by kamsar** [comment]
I've traced this back to a concurrency issue in the Sitecore `TemplateEngine` and have reported it as a bug. Meanwhile, I've set the default concurrency in Unicorn down to 1 thread, which eliminates the issue in my testing (the template engine was designed to be thread safe, sort of - just not under really heavy loads).

To be clear this issue ONLY affects _writing_ of _template items_ and _template fields_, and doing a lo of them at the same time.

Closing for now; in future I may revisit increasing the default thread count.
